src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesMergeIntTest.kt

```diff
@@ -139,6 +139,18 @@ class CaseNotesMergeIntTest : SqsIntegrationTestBase() {
           putRequestedFor(urlPathEqualTo("/mapping/casenotes/merge/from/A1234AA/to/A1234BB")),
         )
       }
+      // ensure the process has finished before the test ends by checking the telemetry
+      await untilAsserted {
+        verify(telemetryClient).trackEvent(
+          eq("casenotes-prisoner-merge"),
+          check {
+            assertThat(it["offenderNo"]).isEqualTo("A1234BB")
+            assertThat(it["removedOffenderNo"]).isEqualTo("A1234AA")
+            assertThat(it["bookingId"]).isEqualTo("1")
+          },
+          isNull(),
+        )
+      }
```

Comment:
Problem was with the next test starting before the service call from the previous one had finished. So the previous call clocked up stub hits making the verify call count wrong.